### ISSUE TITLE
non-primitive scalars (json)

### ISSUE THREAD

**Issue by boredland** [remorses/genql#86]
Hi there!

Lets say I have a custom scalar like this:

```ts
declare type Scalars = {
    Test: {
        k: string;
    };
};
```

When I try to access a property of that scalar, it has this type:

```ts
Pick<{
    k: string;
}, "valueOf"> | undefined
```

Clearly that is not what I wanted, as I now only can use the type-less "valueOf" function. Do you understand why that happens? Is that related to https://github.com/remorses/genql/issues/57 ?

**Comment by remorses** [comment]
Yes, this is currently unsupported

Can you give me what you passed to cli or config to create that scalar declaration? i will take that in consideration when eventually adding this feature

**Comment by remorses** [comment]
Should be fixed in 6.3.0, comment if you still have the issue
